test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -162,43 +209,77 @@ TEST(NoProxyTest, InvalidCIDREntries) {
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.1.0:443");
 }
 
-TEST(ProxyForAddressTest, SetByEnvVar) {
+TEST(ProxyForAddressTest, ChannelArgPreferred) {
   ScopedEnvVar address_proxy(HttpProxyMapper::kAddressProxyEnvVar,
                              "192.168.0.100:2020");
-  auto args = ChannelArgs();
+  auto args = ChannelArgs()
+                  .Set(GRPC_ARG_ADDRESS_HTTP_PROXY, "192.168.0.101:2020")
+                  .Set(GRPC_ARG_ADDRESS_HTTP_PROXY_ENABLED_ADDRESSES,
+                       "255.255.255.255/0");
   auto address = StringToSockaddr("192.168.0.1:3333");
   ASSERT_TRUE(address.ok()) << address.status();
-  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
-  ASSERT_TRUE(mapped.has_value());
-  auto addr_string = grpc_sockaddr_to_string(&mapped.value(), true);
-  ASSERT_TRUE(addr_string.ok()) << addr_string.status();
-  EXPECT_EQ(*addr_string, "192.168.0.100:2020");
+  EXPECT_THAT(HttpProxyMapper().MapAddress(*address, &args),
+              AddressEq("192.168.0.101:2020"));
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.0.1:3333");
 }
 
-TEST(ProxyForAddressTest, ChannelArgPreferred) {
+TEST(ProxyForAddressTest, AddressesNotIncluded) {
   ScopedEnvVar address_proxy(HttpProxyMapper::kAddressProxyEnvVar,
                              "192.168.0.100:2020");
-  auto args =
-      ChannelArgs().Set(GRPC_ARG_ADDRESS_HTTP_PROXY, "192.168.0.101:2020");
-  auto address = StringToSockaddr("192.168.0.1:3333");
+  ScopedEnvVar address_proxy_enabled(
+      HttpProxyMapper::kAddressProxyEnabledAddressesEnvVar,
+      " 192.168.0.0/24 , 192.168.1.1 , 2001:db8:1::0/48 , 2001:db8:2::5");
+  // v4 address
+  auto address = StringToSockaddr("192.168.2.1:3333");
+  ChannelArgs args;
   ASSERT_TRUE(address.ok()) << address.status();
```

Comment:
Please move this up one line.  We should check the value right after we get it.

Reply:
Done